### PR TITLE
[JBIDE 21068] Set maximum length for display name for project wizard

### PR DESCRIPTION
**Pull request by bdshadow**
The limit is, actually, set in swt Text and it is [platform dependent](http://help.eclipse.org/luna/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/swt/widgets/Text.html#LIMIT). In my Fedora 23 
```
public int getTextLimit () {
	checkWidget ();
	if ((style & SWT.MULTI) != 0) return LIMIT;
	int limit = OS.gtk_entry_get_max_length (handle);
	return limit == 0 ? 0xFFFF : untranslateOffset (limit);
}
```
always returns 0xFFFF which is exactly 65536. So couldn't test it in a usual way, but just setting the limit lower for testing purposes.
However, i added ValidationStatus.error when input is bigger so that this check works in another OSs.

### CODE REVIEW

**Comment by jcantrill** [comment]
The maximum limit should be limited to what the max allowed by the server.
Looking at the validation code for project it will be the max length
allowed for an annotation. I'm in my mobile so it's a little hard for me to
provide the value

On Wednesday, May 25, 2016, Dmitry <notifications@github.com> wrote:

> The limit is, actually, set in swt Text and it is platform dependent
> <http://help.eclipse.org/luna/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/swt/widgets/Text.html#LIMIT>.
> In my Fedora 23
>
> public int getTextLimit () {
>     checkWidget ();
>     if ((style & SWT.MULTI) != 0) return LIMIT;
>     int limit = OS.gtk_entry_get_max_length (handle);
>     return limit == 0 ? 0xFFFF : untranslateOffset (limit);
> }
>
> always returns 0xFFFF which is exactly 65536. So couldn't test it in a
> usual way, but just setting the limit lower for testing purposes.
> However, i added ValidationStatus.error when input is bigger so that this
> check works in another OSs.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/jbosstools/jbosstools-openshift/pull/1186
> Commit Summary
>
>    - Set maximum length for display name for project wizard
>
> File Changes
>
>    - *M*
>    plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizardPage.java
>    <https://github.com/jbosstools/jbosstools-openshift/pull/1186/files#diff-0>
>    (5)
>
> Patch Links:
>
>    - https://github.com/jbosstools/jbosstools-openshift/pull/1186.patch
>    - https://github.com/jbosstools/jbosstools-openshift/pull/1186.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jbosstools/jbosstools-openshift/pull/1186>
>


-- 
--
Jeff Cantrill
Senior Software Engineer, Red Hat Engineering
OpenShift Integration Services
Red Hat, Inc.
*Office*: 703-748-4420 | 866-546-8970 ext. 8162420
jcantril@redhat.com
http://www.redhat.com

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizardPage.java [hunk 1]
```diff
@@ -96,6 +98,9 @@ public IStatus validate(Object value) {
 						if(param.contains("\t") || param.contains("\n")) {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizardPage.java]: I would replace "\n" with SWT.LF and "\t" with SWT.TAB

**Comment by adietish** [comment]
Text.setTextLimit(limit) isnt platform specific. Only the constant Text.LIMIT is platform specific, but we dont need this constant anyhow.
And yes, apparently you need to use the server provided maximum lenght for annotations. @jcantrill can you please show @bdshadow how to retrieve that one?

**Comment by bdshadow** [comment]
@jcantrill , @adietish ok, i'll fix it to the maximum length for annotation.
So maybe we need to edit the task description in jira?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizardPage.java (hunk 1 above)
Comment (bdshadow) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizardPage.java]: @adietish  ok

**Comment by adietish** [comment]
@bdshadow ok, added an explanation to the jira description for the fact that the maximum length should get derived from the maximum lenght of OpenShift annotations

**Comment by adietish** [comment]
@bdshadow can you please always use the JBIDE-XXX identifier in your commit message. This way you can instantly see what jira issue a PR is fixing and the same is ture when reading commits in git log.

**Comment by bdshadow** [comment]
@adietish thank you. Yes, ofcourse, i'll fix it

**Comment by fbricon** [comment]
Can you please add some unit tests on the validator, you might need to refactor some code to make it testable

**Comment by bdshadow** [comment]
@jcantrill  and me discussed it. The only limitation for annotations is that its' total length can't exceed 2^63 ([see here](https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/validation/validation.go#L105)).  Web Console also doesn't limit the display name. So int64 in go-lang is the maximum value for display name;
The number is really big. Maybe it's a better to leave the limitation in PR as it is?

@fbricon yes, i'll add it

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/ProjectDisplayNameValidator.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.swt.SWT;
+
+public class ProjectDisplayNameValidator implements IValidator {
+	
+	public static final int DISPLAY_NAME_LENGTH_LIMIT = 65535;
+
+	@Override
+	public IStatus validate(Object value) {
+		if(value != null && !(value instanceof String))
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/ProjectDisplayNameValidator.java]: value != null is redundant with !(value instanceof String)

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/ProjectDisplayNameValidator.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package org.jboss.tools.openshift.internal.ui.validator;
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/ProjectDisplayNameValidator.java]: please add copyright header

File: tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/ui/validator/ProjectDisplayNameValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package org.jboss.tools.openshift.test.ui.validator;
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/ui/validator/ProjectDisplayNameValidatorTest.java]: missing copyright header

**Comment by fbricon** [comment]
Aside from a couple minuscule issues, LGTM. Will merge once @bdshadow updates the PR

**Pull request merged**